File: app/src/main/java/org/stypox/dicio/skills/timer/TimerInfo.java
package org.stypox.dicio.skills.timer;

import static org.stypox.dicio.Sections.getSection;
import static org.stypox.dicio.Sections.isSectionAvailable;
import static org.stypox.dicio.SectionsGenerated.timer;
import static org.stypox.dicio.SectionsGenerated.util_yes_no;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import org.stypox.dicio.R;
import org.dicio.skill.Skill;
import org.dicio.skill.SkillContext;
import org.dicio.skill.SkillInfo;
import org.dicio.skill.chain.ChainSkill;
import org.dicio.skill.standard.StandardRecognizer;

public class TimerInfo extends SkillInfo {

    public TimerInfo() {
        super("timer", R.string.skill_name_timer, R.string.skill_sentence_example_timer,
                R.drawable.ic_timer_white, false);
    }

    @Override
    public boolean isAvailable(final SkillContext context) {
        return isSectionAvailable(timer) && isSectionAvailable(util_yes_no)
                && context.getNumberParserFormatter() != null;
    }

    @Override
    public Skill build(final SkillContext context) {
        return new ChainSkill.Builder()
                .recognize(new StandardRecognizer(getSection(timer)))
                .process(new TimerProcessor())
                .output(new TimerOutput());
    }

    @Nullable
    @Override
    public Fragment getPreferenceFragment() {
        return null;
    }
}

File: app/src/main/kotlin/org/stypox/dicio/skills/timer/TimerInfo.kt
package org.stypox.dicio.skills.timer

import android.content.Context
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Call
import androidx.compose.material.icons.filled.Timer
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.vector.rememberVectorPainter
import androidx.fragment.app.Fragment
import org.dicio.skill.skill.Skill
import org.dicio.skill.context.SkillContext
import org.dicio.skill.skill.SkillInfo
import org.stypox.dicio.R
import org.stypox.dicio.Sections
import org.stypox.dicio.SectionsGenerated.timer
import org.stypox.dicio.SectionsGenerated.util_yes_no
import org.stypox.dicio.sentences.Sentences

object TimerInfo : SkillInfo("timer") {
    override fun name(context: Context) =
        context.getString(R.string.skill_name_timer)

    override fun sentenceExample(context: Context) =
        context.getString(R.string.skill_sentence_example_timer)

    @Composable
    override fun icon() =
        rememberVectorPainter(Icons.Default.Timer)

    override fun isAvailable(ctx: SkillContext): Boolean {
        return Sentences.Timer[ctx.locale.language] != null
                && Sentences.UtilYesNo[ctx.locale.language] != null
                && ctx.parserFormatter != null
    }

    override fun build(ctx: SkillContext): Skill<*> {
        return TimerSkill(TimerInfo, Sentences.Timer[ctx.locale.language]!!)
    }
}
